### PR TITLE
M3-4899: Transfer Details modal

### PR DESCRIPTION
**Pull request by tiffwong**
Opens when the user clicks a token ID from the landing page tables

### CODE REVIEW

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferDetailsDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      onClose={onClose}
+    >
+      <Typography className={classes.token}>
+        <div className={classes.label}>Transfer Token: </div>
+        {token}
+      </Typography>
+      <Typography className={classes.label}>Linode IDs:</Typography>
+      <div className={classes.entities}>
+        {entities?.linodes.map((entity, idx) => {
+          return <Typography data-test-id={idx}>{entity}</Typography>;
```
Comment (Jskobos) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferDetailsDialog.tsx:45]: I think it's `data-testid` if it's for working with React Testing Library.

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferDetailsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        })}
```
Comment (Jskobos) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferDetailsDialog.tsx:46]: Also need a key in here

**Review by Jskobos — CHANGES_REQUESTED** [review]
Looks good, need to fix a missing `key` console error. 

What do you think about borrowing the logic used in the accept transfer modal to also display the time until expiry (if the transfer is pending) in human-readable format ("This transfer will expire in 16 hours (<timestamp>)")?

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferDetailsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <InformationDialog
+      title="Transfer Details"
+      fullWidth
+      open={isOpen}
+      onClose={onClose}
+    >
+      <Typography className={classes.token}>
+        <div className={classes.label}>Transfer Token: </div>
```
Comment (DevDW) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferDetailsDialog.tsx:39]: I think this `<div>` might be causing this console error:

Screen Shot 2021-02-18 at 4 47 43 PM

**Review by Jskobos — APPROVED** [review]

**Review by DevDW — APPROVED** [review]

**Pull request merged**